transformer_engine/pytorch/distributed.py

```diff
@@ -216,7 +216,18 @@ def forward(
             )
 
         # Store everything.
-        ctx.save_for_backward(*args)
+        ctx.inputs = []
+        ctx.tensor_indices = []
+        tensor_inputs = []
+        for i, arg in enumerate(args):
+            if torch.is_tensor(arg):
+                tensor_inputs.append(arg)
+                ctx.tensor_indices.append(i)
+                ctx.inputs.append(None)
+            else:
+                ctx.inputs.append(arg)
+        ctx.save_for_backward(*tensor_inputs)
+
```

Comment:
In order to avoid the Python for loops, I was thinking about something like this instead:
```python
ctx.inputs = [arg if not torch.is_tensor(arg) else None for arg in args]
tensor_inputs = [arg if torch.is_tensor(arg) else None for arg in args]
ctx.save_for_backward(*tensor_inputs)
```
and then you can glue them together with
```python
args  = [t if t is not None else arg for (t, arg) in zip(ctx.saved_tensors, ctx.inputs)]
```
What do you think?

Reply:
This will simplify the codes.
PR modified accordingly.